internal/pkg/cli/local_run.go

```diff
@@ -106,11 +198,205 @@ func (o *localRunOpts) Execute() error {
 	}
 
 	secrets := taskDef.Secrets()
-	_, err = o.ecsLocalClient.DecryptedSecrets(secrets)
+	decryptedSecrets, err := o.ecsLocalClient.DecryptedSecrets(secrets)
 	if err != nil {
 		return fmt.Errorf("get secret values: %w", err)
 	}
 
+	secretsList := make(map[string]string)
+	for _, s := range decryptedSecrets {
+		secretsList[s.Name] = s.Value
+	}
+
+	envVars := make(map[string]string)
+	envVariables := taskDef.EnvironmentVariables()
+	for _, envVariable := range envVariables {
+		envVars[envVariable.Name] = envVariable.Value
+	}
+
+	containerPorts := make(map[string]string)
+	containerdef := taskDef.ContainerDefinitions
+	for _, container := range containerdef {
+		for _, portMapping := range container.PortMappings {
+			hostPort := aws.Int64Value(portMapping.HostPort)
+			hostPortStr := strconv.FormatInt(hostPort, 10)
+			var containerPort int64
+			if portMapping.ContainerPort == nil {
+				containerPort = hostPort
+			} else {
+				containerPort = aws.Int64Value(portMapping.ContainerPort)
+			}
+			containerportStr := strconv.FormatInt(containerPort, 10)
+			containerPorts[hostPortStr] = containerportStr
+		}
+	}
+
+	env, err := o.store.GetEnvironment(o.appName, o.envName)
+	if err != nil {
+		return fmt.Errorf("get environment %q configuration: %w", o.envName, err)
+	}
+	o.targetEnv = env
+
+	app, err := o.store.GetApplication(o.appName)
+	if err != nil {
+		return fmt.Errorf("get application %q configuration: %w", o.appName, err)
+	}
+	o.targetApp = app
+
+	envSess, err := o.sessProvider.FromRole(env.ManagerRoleARN, env.Region)
+	if err != nil {
+		return fmt.Errorf("get env session: %w", err)
+	}
+	mft, err := workloadManifest(&workloadManifestInput{
+		name:         o.wkldName,
+		appName:      o.appName,
+		envName:      o.envName,
+		interpolator: o.newInterpolator(o.appName, o.envName),
+		ws:           o.ws,
+		unmarshal:    o.unmarshal,
+		sess:         envSess,
+	})
+	if err != nil {
+		return err
+	}
+	o.appliedDynamicMft = mft
+	o.repository, err = o.configureClients(o)
+	if err != nil {
+		return err
+	}
+
+	err = o.buildContainerImages(o)
+	if err != nil {
+		return fmt.Errorf("building container image: %w", err)
+	}
+
+	containerNames := o.out.ContainerNames
+	imageNames := o.out.ImageNames
+
+	for i, image := range imageNames {
+		o.imageInfoList = append(o.imageInfoList, imageInfo{
+			containerName: containerNames[i],
+			imageURI:      image,
+		})
+	}
```

Comment:
I see we are assuming here that each item in `o.out.ContainerNames` correspond to the item with the same index in `o.out.ImageURI`. This protocol isn't established anywhere, and can only be inferred by looking at the code in `cli/deploy` (which is a different package than where we are now). This is unfortunately error-prone in future development : (. I think we could just have a struct
```go
// in cli/deploy
type ImagePerContainer struct {
 ContainerName string
 ImageURI string
}
```

or is it an option to embed these information into `ContainerImageIdentifier`?